tests/test_equipment.py

```diff
@@ -0,0 +1,414 @@
+"""This file contains all the tests to do with Equipment."""
+import os
+from pathlib import Path
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from tlo import Date, Module, Simulation
+from tlo.analysis.utils import parse_log_file
+from tlo.events import IndividualScopeEventMixin
+from tlo.methods import Metadata, demography, healthsystem
+from tlo.methods.equipment import Equipment
+from tlo.methods.hsi_event import HSI_Event
+
+resourcefilepath = Path(os.path.dirname(__file__)) / "../resources"
+
+
+def test_core_functionality_of_equipment_class(seed):
```

Comment:
As a general comment on this test function - the checks look sensible here, but generally I would say better to have lots of smaller test functions with only a few assert statements rather than one big test function with lots of asserts as this means you get less granular information on test failure (as if one of the earlier asserts fails none of the subsequent asserts are run which means you have less information available for finding out the source of errors). To avoid having to repeatedly construct the `Equipment` class instance you could make this (and its arguments) fixtures, that is something like the following

```Python
@pytest.fixture
def catalogue():
    return pd.DataFrame(
        [
            {"Item_Description": "ItemZero", "Item_Code": 0, "Pkg_Name": float('nan')},
            {"Item_Description": "ItemOne", "Item_Code": 1, "Pkg_Name": float('nan')},
            {"Item_Description": "ItemTwo", "Item_Code": 2, "Pkg_Name": 'PkgWith2+3'},
            {"Item_Description": "ItemThree", "Item_Code": 3, "Pkg_Name": 'PkgWith2+3'},
        ]
    )

@pytest.fixture
def data_availability():
    return pd.DataFrame(
        # item 0 is not available anywhere; item 1 is available everywhere; item 2 is available only at facility_id=1
        # No data for fac_id=2
        [
            {"Item_Code": 0, "Facility_ID": 0, "Pr_Available": 0.0},
            {"Item_Code": 0, "Facility_ID": 1, "Pr_Available": 0.0},
            {"Item_Code": 1, "Facility_ID": 0, "Pr_Available": 1.0},
            {"Item_Code": 1, "Facility_ID": 1, "Pr_Available": 1.0},
            {"Item_Code": 2, "Facility_ID": 0, "Pr_Available": 0.0},
            {"Item_Code": 2, "Facility_ID": 1, "Pr_Available": 1.0},
        ]
    )

@pytest.fixture
def master_facilities_list():
    return pd.DataFrame(
        [
            {'District': 'D0', 'Facility_Level': '1a', 'Region': 'R0', 'Facility_ID': 0, 'Facility_Name': 'Fac0'},
            {'District': 'D0', 'Facility_Level': '1b', 'Region': 'R0', 'Facility_ID': 1, 'Facility_Name': 'Fac1'},
            {'District': 'D0', 'Facility_Level': '2', 'Region': 'R0', 'Facility_ID': 2, 'Facility_Name': 'Fac2'},
        ]
    )

@pytest.fixture
def rng(seed):
    return np.random.RandomState(np.random.MT19937(np.random.SeedSequence(seed)))

@pytest.fixture
def equipment_default(catalogue, data_availability, master_facilities_list, rng):
    return Equipment(
        catalogue=catalogue,
        data_availability=data_availability,
        rng=rng,
        master_facilities_list=master_facilities_list,
        availability="default",
    )

@pytest.fixture
def equipment_default(catalogue, data_availability, master_facilities_list, rng):
    return Equipment(
        catalogue=catalogue,
        data_availability=data_availability,
        rng=rng,
        master_facilities_list=master_facilities_list,
        availability="default",
    )

@pytest.fixture
def equipment_all(catalogue, data_availability, master_facilities_list, rng):
    return Equipment(
        catalogue=catalogue,
        data_availability=data_availability,
        rng=rng,
        master_facilities_list=master_facilities_list,
        availability="all",
    )

def test_parse_single_item_code(equipment_default):
    assert {1} == equipment_default.parse_items(1)

def test_parse_iterable_of_item_codes(equipment_default):
    assert {1, 2} == equipment_default.parse_items([1, 2])

def test_parse_single_item_descriptor(equipment_default):
    assert {1} == equipment_default.parse_items('ItemOne')

def test_parse_iterable_of_item_descriptors(equipment_default):
    assert {1, 2} == equipment_default.parse_items(['ItemOne', 'ItemTwo'])

@pytest.mark.parametrize("container_type", [list, tuple, set])
def test_parse_empty_items(equipment_default, container_type):
    assert set() == eq_default.parse_items(container_type())

def test_parse_unrecognised_item_code_warns(equipment_default):
    with pytest.warns():
        equipment_default.parse_items(10001)

def test_parse_unrecognised_item_descriptor_warns(equipment_default):
    with pytest.warns():
        equipment_default.parse_items('ItemThatIsNotDefined')

...
```

Possibly not worth the effort doing this here to avoid delaying getting this PR merged in further, but thought it might be useful to note for future!

Reply:
Thanks Matt - that's really useful to know. As you guessed, I don't think I want to make that change to these tests now (though I am tempted as you've made it look nice and easy), in the interest of getting this in and carrying on with defining the actual items... but it's certainly noted for next time. I can certainly see the advantages of doing it this way.